doc/plutus-core-spec/untyped-cek-machine.tex

```diff
@@ -154,30 +154,33 @@ \section{The CEK machine}
                          \inConstrFrame{i}{\repetition{V} \cons V}{(\repetition{M}, \rho)} \cons s;\rho \compute M \\
 \inConstrFrame{i}{\repetition{V}}{([], \rho)} \cons s & \return V   &~\mapsto~&
                          s \return \VConstr{i}{\repetition{V} \cons V} \\
-\inCaseFrame{(M_0 \ldots M_n, \rho)} \cons s & \return \VConstr{i}{V_0 \ldots V_m}   &~\mapsto~&
-                         \inAppLeftFrame{V_m} \cons \cdots \cons \inAppLeftFrame{V_0} \cons s ;\rho \compute M_i \enskip \text{if $0 \leq i \leq n$}
+\inCaseFrame{(M_0 \ldots M_n, \rho)} \cons s & \return \VConstr{i}{V_1 \ldots V_m}   &~\mapsto~&
+                         \inAppLeftFrame{V_m} \cons \cdots \cons \inAppLeftFrame{V_1} \cons s ;\rho \compute M_i \enskip \text{if $0 \leq i \leq n$}
 \end{alignat*}
 \end{minipage}
     \caption{CEK machine transitions for Plutus Core}
     \label{fig:untyped-cek-transitions}
 \end{subfigure}
 
 \bigskip
-  \begin{subfigure}[c]{\linewidth}
-$$ \EvalCEK(s, b, [V_1, \ldots, V_n]) =
-   \begin{cases}
-      \cekerror  & \text{if $\Eval\,(b,[V_1, \ldots, V_n]) = \errorX$}\\
-      s \return \Eval\,(b,[V_1, \ldots, V_n]) & \text{otherwise}
-   \end{cases}
-$$
-    \caption{Evaluation of built-in functions}
-    \label{fig:untyped-cek-builtins}
-    \end{subfigure}
+\begin{subfigure}[c]{\linewidth}
+  \begin{align*}
+ \EvalCEK(s, b, [V_1, \ldots, V_n]) =
+  &   \begin{cases}
+        \cekerror  & \text{if $r = \errorX$}\\
+        \inAppLeftFrame{V^{\prime}_m} \cons \cdots \cons \inAppLeftFrame{V^{\prime}_1} \cons s \return V^{\prime}
+          & \text{if $r = (V^{\prime}|V^{\prime}_1,\ldots,V^{\prime}_m) \in \R^{+}$}
```

Comment:
I was briefly worried about the order here because it looks as if this is doing the same thing as `case` (see the bottom line of the main machine specifcation) but in the code you have to reverse the list of arguments to `constr`.  This is OK because in the second-last line of the figure the arguments are appended to the end of thelist, not the start.